frontend/src/components/Refresh/RefreshNotifier.tsx

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from "react";
+import { TimeInMilliseconds } from "../../types/Common";
+import useRefreshInterval from "../../hooks/refresh";
+
+interface Props {
+  onTick: (timestamp: TimeInMilliseconds) => void;
+}
+
+export default function RefreshNotifier({ onTick }: Props) {
```

Comment:
It will call the `onTick` function that is passed in the arguments when a refresh is triggered. It is its only goal :-)

As you commented in your review comment, this `RefreshNotifier` is intended to be used in class components, just to prevent a refactor to a function component. It won't render nothing, but it still invokes that `onTick` callback.

Reply:
With this, do we need the `connectRefresh` HOC at all then? Aren't they serving the same purpose?

You mind adding some comments here explaining this component's purpose?

Reply:
All three - the `useRefreshInterval` hook, the `RefreshNotifier` and the `connectRefresh` HOC - serve the same purpose. The hook and the `RefreshNotifier` would have been enough. I added the HOC just to avoid a major refactor in the class components, which have large logic in the `componentDidUpdate` to find when to refresh data.

Reply:
> You mind adding some comments here explaining this component's purpose?

I'll do that :-)

Reply:
Having read through the other comments as well, please be generous with in-code comments for this PR, so that future devs can have a head start understanding why things are like they will be after this is merged.

Reply:
I added a comment. Not sure if that will be enough. You tell :-)